cfme/intelligence/optimization.py

```diff
@@ -0,0 +1,157 @@
+import attr
+from cached_property import cached_property
+from navmazing import NavigateToAttribute
+from widgetastic.widget import Table
+from widgetastic.widget import Text
+from widgetastic_patternfly import Button
+from widgetastic_patternfly import SelectorDropdown
+
+from cfme.common import BaseLoggedInPage
+from cfme.intelligence.reports.reports import ReportsCollection
+from cfme.intelligence.reports.saved import SavedReportsCollection
+from cfme.modeling.base import BaseCollection
+from cfme.modeling.base import BaseEntity
+from cfme.utils.appliance.implementations.ui import CFMENavigateStep
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.appliance.implementations.ui import navigator
+from cfme.utils.wait import wait_for
+from widgetastic_manageiq import SearchBox
+
+
+class OptimizationView(BaseLoggedInPage):
+    title = Text('//*[@id="main-content"]//h1')
+    table = Table(
+        '//*[@id="main_div"]//table', column_widgets={"Action": Button("contains", "Queue Report")}
+    )
+    refresh = Button(title="Refresh the list")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.logged_in_as_current_user
+            and self.title.text == "Optimization"
+            and self.table.is_displayed
+            and self.navigation.currently_selected == ["Overview", "Optimization"]
+            and self.breadcrumb.locations == ["Overview", "Optimization"]
+        )
+
+
+class OptimizationReportAllView(OptimizationView):
+    table = Table('//*[@id="main_div"]//table')
+
+    @property
+    def is_displayed(self):
+        menu_name = self.context["object"].menu_name
+        return (
+            self.logged_in_as_current_user
+            and self.title.text == menu_name
+            and self.navigation.currently_selected == ["Overview", "Optimization"]
+            and self.breadcrumb.locations == ["Overview", "Optimization", menu_name]
+        )
+
+
+class OptimizationSavedReportDetailsView(OptimizationView):
+    field = SelectorDropdown("id", "filterFieldTypeMenu")
+    search_text = SearchBox(locator='//input[contains(@placeholder, "search text")]')
+    table = Table('//*[@id="main_div"]//table')
+
+    @property
+    def is_displayed(self):
+        context_obj = self.context["object"]
+        return (
+            self.logged_in_as_current_user
+            and self.title.text == context_obj.name
+            and self.table.is_displayed
+            and self.navigation.currently_selected == ["Overview", "Optimization"]
+            and self.breadcrumb.locations
+            == ["Overview", "Optimization", context_obj.parent_obj.menu_name, context_obj.name]
+        )
+
+
+@attr.s
+class OptimizationReport(BaseEntity):
+    menu_name = attr.ib()
+    last_run_at = attr.ib(default=None)
+    runs = attr.ib(default=None)
+
+    def __attrs_post_init__(self):
+        self._collections = {
```

Comment:
This should not be done in `__attrs_post_init__`, the Base model classes handle this via direct definition within the class.

https://cfme-tests.readthedocs.io/en/master/guides/collections.html#changes-to-previous-model-versions